ikarus/controlroutines/adaptivestepsizing.hh

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: 2021-2023 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+#include <ikarus/controlroutines/pathfollowingfunctions.hh>
+#include <ikarus/solver/nonlinearsolver/solverinfos.hh>
+
+namespace Ikarus::AdaptiveStepSizing {
+  template <typename ScalarType = double>
+  struct NoOp {
+  public:
+    template <typename NonLinearOperator>
+    void operator()(const NonLinearSolverInformation&, Ikarus::SubsidiaryArgs&, const NonLinearOperator&) {}
+    [[nodiscard]] int targetIterations() const { return 0; }
+    void setTargetIterations(int) {}
+  };
+
+  template <typename ScalarType = double>
+  struct IterationBased {
+  public:
+    template <typename NonLinearOperator>
+    void operator()(const NonLinearSolverInformation& solverInfo, Ikarus::SubsidiaryArgs& subsidiaryArgs,
+                    const NonLinearOperator&) {
+      if (not((subsidiaryArgs.actualStep > 0) and (targetIterations_ != 0)))
```

Comment:
Can you seperate these two if statements? Currently this is not readable
The structure of the code should be imho:

```cpp
if (subsidiaryArgs.actualStep == 0)
return;

if ( targetIterations_ == 0)
 DUNE_THROW(...);

 // do the usual stuff

```

Reply:
Agreed and modified